web/core/components/issues/issue-detail-widgets/relations/content.tsx

```diff
@@ -0,0 +1,182 @@
+"use client";
+import { FC, useState } from "react";
+import { observer } from "mobx-react";
+import { CircleDot, CopyPlus, XCircle } from "lucide-react";
+import { TIssue, TIssueRelationIdMap } from "@plane/types";
+import { Collapsible, RelatedIcon } from "@plane/ui";
+// components
+import { RelationIssueList } from "@/components/issues";
+import { DeleteIssueModal } from "@/components/issues/delete-issue-modal";
+import { CreateUpdateIssueModal } from "@/components/issues/issue-modal";
+// hooks
+import { useIssueDetail } from "@/hooks/store";
+// helper
+import { useRelationOperations } from "./helper";
+
+type Props = {
+  workspaceSlug: string;
+  projectId: string;
+  issueId: string;
+  disabled: boolean;
+};
+
+const ISSUE_RELATION_OPTIONS = [
+  {
+    key: "blocked_by",
+    label: "Blocked by",
+    icon: (size: number) => <CircleDot size={size} />,
+    className: "bg-red-500/20 text-red-700",
+  },
+  {
+    key: "blocking",
+    label: "Blocking",
+    icon: (size: number) => <XCircle size={size} />,
+    className: "bg-yellow-500/20 text-yellow-700",
+  },
+  {
+    key: "relates_to",
+    label: "Relates to",
+    icon: (size: number) => <RelatedIcon height={size} width={size} />,
+    className: "bg-custom-background-80 text-custom-text-200",
+  },
+  {
+    key: "duplicate",
+    label: "Duplicate of",
+    icon: (size: number) => <CopyPlus size={size} />,
+    className: "bg-custom-background-80 text-custom-text-200",
+  },
+];
+
+type TIssueCrudState = { toggle: boolean; issueId: string | undefined; issue: TIssue | undefined };
+
+export const RelationsCollapsibleContent: FC<Props> = observer((props) => {
+  const { workspaceSlug, projectId, issueId, disabled = false } = props;
+  // state
+  const [issueCrudState, setIssueCrudState] = useState<{
+    update: TIssueCrudState;
+    delete: TIssueCrudState;
+  }>({
```

Comment:
Can we have different states for the objects?

Reply:
A ticket has been raised for this [WEB-1861](https://app.plane.so/plane/projects/02c3e1d5-d7e2-401d-a773-45ecba45d745/issues/8caad235-0115-47ed-82ae-0112cbe1db67/). I'll take care of it.